com.hazelcast.jet.kinesis.KinesisFailureTest.com.hazelcast.jet.kinesis.KinesisFailureTest

The same for **com.hazelcast.jet.kinesis.KinesisIntegrationTest.com.hazelcast.jet.kinesis.KinesisIntegrationTest** test

_master_5.1_ (commit 0f734dd0617deedb4b9a69598206075ff758042f)

Failed on MacOS, Zulu, JDK 8: [test failures report](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-MacOS-ZuluJDK8/15/testReport/)

<details><summary>Stacktrace:</summary>

```
java.lang.ExceptionInInitializerError
	at sun.misc.Unsafe.ensureClassInitialized(Native Method)
	at sun.reflect.UnsafeFieldAccessorFactory.newFieldAccessor(UnsafeFieldAccessorFactory.java:43)
	at sun.reflect.ReflectionFactory.newFieldAccessor(ReflectionFactory.java:156)
	at java.lang.reflect.Field.acquireFieldAccessor(Field.java:1088)
	at java.lang.reflect.Field.getFieldAccessor(Field.java:1069)
	at java.lang.reflect.Field.get(Field.java:393)
	at org.junit.runners.model.FrameworkField.get(FrameworkField.java:92)
	at org.junit.runners.model.TestClass.collectAnnotatedFieldValues(TestClass.java:249)
	at org.junit.runners.ParentRunner.classRules(ParentRunner.java:280)
	at org.junit.runners.ParentRunner.withClassRules(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.classBlock(ParentRunner.java:217)
	at com.hazelcast.test.AbstractParameterizedHazelcastClassRunner.classBlock(AbstractParameterizedHazelcastClassRunner.java:139)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:412)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:109)
	at org.testcontainers.DockerClientFactory.getOrInitializeStrategy(DockerClientFactory.java:136)
	at org.testcontainers.DockerClientFactory.getTransportConfig(DockerClientFactory.java:142)
	at org.testcontainers.DockerClientFactory.getRemoteDockerUnixSocketPath(DockerClientFactory.java:152)
	at org.testcontainers.containers.localstack.LocalStackContainer.<init>(LocalStackContainer.java:96)
	at org.testcontainers.containers.localstack.LocalStackContainer.<init>(LocalStackContainer.java:82)
	at com.hazelcast.jet.kinesis.KinesisIntegrationTest.<clinit>(KinesisIntegrationTest.java:74)
	... 32 more
```

</details>

<details><summary>Standard output:</summary>

```
Started Running Test: serializable_classes_should_have_valid_serialVersionUID
```

</details>

--- a/extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisFailureTest.java
+++ b/extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisFailureTest.java
@@ -74,12 +74,9 @@ public class KinesisFailureTest extends AbstractKinesisTest {
     }
 
     @BeforeClass
-    public static void beforeClassCheckDocker() {
-        assumeTrue(DockerClientFactory.instance().isDockerAvailable());
-    }
-
-    @BeforeClass
     public static void beforeClass() {
+        assumeTrue(DockerClientFactory.instance().isDockerAvailable());
+
         localStack = new LocalStackContainer(parse("localstack/localstack")
                 .withTag("0.12.3"))
                 .withNetwork(NETWORK)
@@ -107,7 +104,9 @@ public class KinesisFailureTest extends AbstractKinesisTest {
 
     @AfterClass
     public static void afterClass() {
-        KINESIS.shutdown();
+        if (KINESIS != null) {
+            KINESIS.shutdown();
+        }
 
         if (toxiProxy != null) {
             toxiProxy.stop();

--- a/extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisFailureTest.java
+++ b/extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisFailureTest.java
@@ -60,17 +60,9 @@ public class KinesisFailureTest extends AbstractKinesisTest {
     @ClassRule
     public static final Network NETWORK = Network.newNetwork();
 
-    @ClassRule
-    public static final LocalStackContainer LOCALSTACK = new LocalStackContainer(parse("localstack/localstack")
-            .withTag("0.12.3"))
-            .withNetwork(NETWORK)
-            .withServices(Service.KINESIS);
+    public static LocalStackContainer localStack;
 
-    @ClassRule
-    public static final ToxiproxyContainer TOXIPROXY = new ToxiproxyContainer(parse("shopify/toxiproxy")
-            .withTag("2.1.0"))
-            .withNetwork(NETWORK)
-            .withNetworkAliases("toxiproxy");
+    public static ToxiproxyContainer toxiProxy;
 
     private static AwsConfig AWS_CONFIG;
     private static AmazonKinesisAsync KINESIS;
@@ -88,16 +80,27 @@ public class KinesisFailureTest extends AbstractKinesisTest {
 
     @BeforeClass
     public static void beforeClass() {
+        localStack = new LocalStackContainer(parse("localstack/localstack")
+                .withTag("0.12.3"))
+                .withNetwork(NETWORK)
+                .withServices(Service.KINESIS);
+        localStack.start();
+        toxiProxy = new ToxiproxyContainer(parse("shopify/toxiproxy")
+                .withTag("2.1.0"))
+                .withNetwork(NETWORK)
+                .withNetworkAliases("toxiproxy");
+        toxiProxy.start();
+
         System.setProperty(SDKGlobalConfiguration.AWS_CBOR_DISABLE_SYSTEM_PROPERTY, "true");
         // with the jackson versions we use (2.11.x) Localstack doesn't without disabling CBOR
         // https://github.com/localstack/localstack/issues/3208
 
-        PROXY = TOXIPROXY.getProxy(LOCALSTACK, 4566);
+        PROXY = toxiProxy.getProxy(localStack, 4566);
 
         AWS_CONFIG = new AwsConfig()
                 .withEndpoint("http://" + PROXY.getContainerIpAddress() + ":" + PROXY.getProxyPort())
-                .withRegion(LOCALSTACK.getRegion())
-                .withCredentials(LOCALSTACK.getAccessKey(), LOCALSTACK.getSecretKey());
+                .withRegion(localStack.getRegion())
+                .withCredentials(localStack.getAccessKey(), localStack.getSecretKey());
         KINESIS = AWS_CONFIG.buildClient();
         HELPER = new KinesisTestHelper(KINESIS, STREAM, Logger.getLogger(KinesisIntegrationTest.class));
     }
@@ -105,6 +108,13 @@ public class KinesisFailureTest extends AbstractKinesisTest {
     @AfterClass
     public static void afterClass() {
         KINESIS.shutdown();
+
+        if (toxiProxy != null) {
+            toxiProxy.stop();
+        }
+        if (localStack != null) {
+            localStack.stop();
+        }
     }
 
     @Test

--- a/extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisIntegrationTest.java
+++ b/extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisIntegrationTest.java
@@ -69,10 +69,7 @@ import static org.testcontainers.utility.DockerImageName.parse;
 
 public class KinesisIntegrationTest extends AbstractKinesisTest {
 
-    @ClassRule
-    public static final LocalStackContainer LOCALSTACK = new LocalStackContainer(parse("localstack/localstack")
-            .withTag("0.12.3"))
-            .withServices(Service.KINESIS);
+    public static LocalStackContainer localStack;
 
     private static AwsConfig AWS_CONFIG;
     private static AmazonKinesisAsync KINESIS;
@@ -90,6 +87,11 @@ public class KinesisIntegrationTest extends AbstractKinesisTest {
 
     @BeforeClass
     public static void beforeClass() {
+        localStack = new LocalStackContainer(parse("localstack/localstack")
+                .withTag("0.12.3"))
+                .withServices(Service.KINESIS);
+        localStack.start();
+
         // To run with real kinesis AWS credentials need be available
         // to be loaded by DefaultAWSCredentialsProviderChain.
         // Keep in mind the real Kinesis is paid service and once you
@@ -106,9 +108,9 @@ public class KinesisIntegrationTest extends AbstractKinesisTest {
                     .withRegion("us-east-1");
         } else {
             AWS_CONFIG = new AwsConfig()
-                    .withEndpoint("http://" + LOCALSTACK.getHost() + ":" + LOCALSTACK.getMappedPort(4566))
-                    .withRegion(LOCALSTACK.getRegion())
-                    .withCredentials(LOCALSTACK.getAccessKey(), LOCALSTACK.getSecretKey());
+                    .withEndpoint("http://" + localStack.getHost() + ":" + localStack.getMappedPort(4566))
+                    .withRegion(localStack.getRegion())
+                    .withCredentials(localStack.getAccessKey(), localStack.getSecretKey());
         }
         KINESIS = AWS_CONFIG.buildClient();
         HELPER = new KinesisTestHelper(KINESIS, STREAM, Logger.getLogger(KinesisIntegrationTest.class));
@@ -117,6 +119,10 @@ public class KinesisIntegrationTest extends AbstractKinesisTest {
     @AfterClass
     public static void afterClass() {
         KINESIS.shutdown();
+
+        if (localStack != null) {
+            localStack.stop();
+        }
     }
 
     @Test

--- a/extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisIntegrationTest.java
+++ b/extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisIntegrationTest.java
@@ -36,7 +36,6 @@ import com.hazelcast.logging.Logger;
 import com.hazelcast.test.annotation.NightlyTest;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
-import org.junit.ClassRule;
 import org.junit.Ignore;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;

--- a/extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisIntegrationTest.java
+++ b/extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisIntegrationTest.java
@@ -80,12 +80,9 @@ public class KinesisIntegrationTest extends AbstractKinesisTest {
     }
 
     @BeforeClass
-    public static void beforeClassCheckDocker() {
-        assumeTrue(DockerClientFactory.instance().isDockerAvailable());
-    }
-
-    @BeforeClass
     public static void beforeClass() {
+        assumeTrue(DockerClientFactory.instance().isDockerAvailable());
+
         localStack = new LocalStackContainer(parse("localstack/localstack")
                 .withTag("0.12.3"))
                 .withServices(Service.KINESIS);
@@ -117,7 +114,9 @@ public class KinesisIntegrationTest extends AbstractKinesisTest {
 
     @AfterClass
     public static void afterClass() {
-        KINESIS.shutdown();
+        if (KINESIS != null) {
+            KINESIS.shutdown();
+        }
 
         if (localStack != null) {
             localStack.stop();

